start.go

```diff
@@ -410,6 +425,22 @@ func getCtx(svrCtx *server.Context, block bool) (*errgroup.Group, context.Contex
 	return g, ctx
 }
 
+// getAppVersion returns the application version to be used.
+func getAppVersion(svrCtx *server.Context, appCreator types.AppCreator) (uint64, error) {
```

Comment:
it will always use the in process app, `appCreator` is the latest app. Not sure it will work.